grace/utils/augment_graph.py

```diff
@@ -0,0 +1,124 @@
+from typing import Any, Dict, Tuple
+
+import torch
+
+import numpy as np
+import networkx as nx
+
+from grace.base import GraphAttrs, Annotation
+
+
+def find_average_annotation(
+    edge: Tuple[int],
+    graph: nx.Graph,
+) -> Annotation:
+    """Finds the average annotation of edges connected to
+    the nodes defining an input edge.
+
+    Parameters
+    ----------
+    edge : Tuple[int]
+        Tuple of nodes that define the edge
+
+    Returns
+    -------
+    annotation : GraphAttrs.EDGE_GROUND_TRUTH
+        Calculated annotation for the edge
+    """
+
+    annotation_list = [
+        connection[GraphAttrs.EDGE_GROUND_TRUTH]
+        for node in edge
+        for connection in graph.adj[node].values()
+    ]
+    num_positive = annotation_list.count(Annotation.TRUE_POSITIVE)
+    num_negative = annotation_list.count(Annotation.TRUE_NEGATIVE)
+
+    if num_positive > num_negative:
+        return Annotation.TRUE_POSITIVE
+    elif num_negative > num_positive:
+        return Annotation.TRUE_NEGATIVE
+    else:
+        return Annotation.UNKNOWN
+
+
+ANNOTATION_MODES = {"random", "unknown", "average"}
```

Comment:
For consistency I would suggest making this an Enum (like `Annotation` or `GraphAttrs`)